acceptance/tests/ticket_4622_filebucket_diff_test.rb

```diff
@@ -1,86 +1,72 @@
 test_name "ticket 4622 filebucket diff test."
 confine :except, :platform => 'windows'
+skip_test 'skip test, no non-Windows agents specified' if agents.empty?
 
-def create_testfile(filename, contents)
-  create_remote_file(agent, filename, "#{contents}")
-end
-
-def get_checksum_from_backup(filename, bucket_locale)
-  on agent, puppet("filebucket backup #{filename} #{bucket_locale}"), :acceptable_exit_codes => [ 0, 2 ]
+def get_checksum_from_backup_on(host, filename, bucket_locale)
+  on host, puppet("filebucket backup #{filename} #{bucket_locale}"), :acceptable_exit_codes => [ 0, 2 ]
   output = result.stdout.strip.split(": ")
   checksum = output.last
   return checksum
 end
 
-def validate_diff(item_one, item_two, bucket_locale)
-  on agent, puppet("filebucket diff #{item_one} #{item_two} #{bucket_locale}"), :acceptable_exit_codes => [ 0, 2 ]
+def validate_diff_on(host, item_one, item_two, bucket_locale)
+  on host, puppet("filebucket diff #{item_one} #{item_two} #{bucket_locale}"), :acceptable_exit_codes => [ 0, 2 ]
   assert_match("-foo", result.stdout.strip)
   assert_match('+bar', result.stdout.strip)
   assert_match('+baz', result.stdout.strip)
```

Comment:
Since we are excluding Windows, we could use patch to update the file based on a defined diff and use the diff to do the validation as well.
```
create_remote_file_on(host, tmpfile,"foo")
patch =<< EOS 
--- a/#{tmpfile}
+++ b/#{tmpfile}
@@ -1 +1,2 @@
-foo
+bar
+baz
EOS
patchfile = host.tmpfile('patchfile')
create_remote_file_on(host, patchfile,patch)
on host, "patch #{tmpfile} #{patchfile}"
```